torch/csrc/jit/passes/quantization.cpp

```diff
@@ -514,51 +511,48 @@ void InsertQuantDeQuantImpl(
     }
   }
 
-  std::vector<Value*> values_to_quantize;
-  std::unordered_map<script::ModulePtr, script::Module>
-      child_modules_to_quantize;
   QuantizeHelper qh(module);
   std::stack<Block*> blocks_to_visit;
   blocks_to_visit.push(graph->block());
   while (!blocks_to_visit.empty()) {
     Block* b = blocks_to_visit.top();
     blocks_to_visit.pop();
-    for (Node* n : b->nodes()) {
+    for (auto it = b->nodes().begin(), end = b->nodes().end(); it != end;) {
+      Node* n = *it++;
       for (Value* v : n->outputs()) {
         if (v->type()->isSubtypeOf(TensorType::get())) {
-          auto child_module = qh.findChildModuleToQuantize(v);
-          if (child_module) {
-            child_modules_to_quantize[child_module.value().module_object()] =
-                child_module.value();
+          if (v->node()->kind() == prim::CallMethod) {
+            auto module_instance = v->node()->inputs()[0];
+            auto module_method_name = v->node()->s(attr::name);
+            // calling method on self
+            if (module_instance == graph->inputs()[0]) {
+              InsertQuantDeQuantImpl(module, module_method_name);
+            } else {
+              auto child_module = qh.findChildModuleToQuantize(module_instance);
+              if (child_module) {
+                InsertQuantDeQuantImpl(child_module.value(), module_method_name);
```

Comment:
Nit: might be worth moving `InsertQuantDeQuantImpl` after the if so that there is only one place where we invoke it. Then the code before it can be (optionally) factored out to a separate function.